src/credentials/wizards/connections.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import { getIcon, codicon } from '../../shared/icons'
+import { telemetry } from '../../shared/telemetry/telemetry'
+import { createRefreshButton, createExitButton } from '../../shared/ui/buttons'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { Auth, Connection, isIamConnection, reauthCommand } from '../auth'
+import { getDependentAuths } from '../secondaryAuth'
+
+export const getConnectionIcon = (conn: Connection) =>
+    conn.type === 'sso' ? getIcon('vscode-account') : getIcon('vscode-key')
+
+export function createConnectionPrompter(auth: Auth, type?: 'iam' | 'sso') {
+    const placeholder =
+        type === 'iam'
+            ? localize('aws.auth.promptConnection.iam.placeholder', 'Select an IAM credential')
+            : localize('aws.auth.promptConnection.all.placeholder', 'Select a connection')
+
+    const refreshButton = createRefreshButton()
+    refreshButton.onClick = () => void prompter.clearAndLoadItems(loadItems())
+
+    const prompter = createQuickPick(loadItems(), {
+        placeholder,
+        title: localize('aws.auth.promptConnection.title', 'Switch Connection'),
+        buttons: [refreshButton, createExitButton()],
+    })
+
+    return prompter
+
+    async function loadItems(): Promise<DataQuickPickItem<Connection | 'addNewConnection' | 'editCredentials'>[]> {
```

Comment:
Placing function declarations _after_ the return statement of the outer scope is common in some codebases. Not sure if I like it or not though.